initfiles/componentfiles/configschema/xsd/esp.xsd

```diff
@@ -145,91 +214,176 @@
+                            <xs:attribute name="cipherList" type="xs:string" use="optional"
+                                          hpcc:presetValue="ECDH+AESGCM:DH+AESGCM:ECDH+AES256:DH+AES256:ECDH+AES128:DH+AES:ECDH+3DES:DH+3DES:RSA+AESGCM:RSA+AES:RSA+3DES:!aNULL:!MD5"
+                                          hpcc:displayName="Cipher List"
+                                          hpcc:tooltip="Sets the ordered list of available ciphers for use by openssl.  See openssl documentation on ciphers for information on use and formatting."/>
+                            <xs:attribute name="passphrase" type="xs:string" use="optional"
+                                          hpcc:modifiers="mask,verify,ignoreNoChange,encrypt"
+                                          hpcc:displayName="Passphrase"
+                                          hpcc:tooltip="The passphrase used to generate the private key"/>
+                            <xs:attribute name="privateKeyFileName" type="xs:string" use="optional"
+                                          hpcc:presetValue="privatekey.cer" hpcc:displayName="Private Key Filename"
+                                          hpcc:requiredIf="../EspBinding[@protocol='https']"
+                                          hpcc:tooltip="Name of destination file in which the private key will be written"/>
                             <!-- The following are deprecated and marked hidden so that they don't appear to the user -->
-                            <xs:attribute name="city" hpcc:deprecated="true" hpcc:hidden="true" type="xs:string" use="optional"/>
-                            <xs:attribute name="country" hpcc:deprecated="true" hpcc:hidden="true" type="xs:string" use="optional"/>
-                            <xs:attribute name="daysValid" hpcc:deprecated="true" hpcc:hidden="true" type="xs:string" use="optional"/>
-                            <xs:attribute name="enableVerification" hpcc:deprecated="true" hpcc:hidden="true" type="xs:boolean" use="optional"/>
-                            <xs:attribute name="organization" hpcc:deprecated="true" hpcc:hidden="true" type="xs:string" use="optional"/>
-                            <xs:attribute name="organizationalUnit" hpcc:deprecated="true" hpcc:hidden="true" type="xs:string" use="optional"/>
-                            <xs:attribute name="regenerateCredentials" hpcc:deprecated="true" hpcc:hidden="true" type="xs:boolean" use="optional"/>
-                            <xs:attribute name="requireAddressMatch" hpcc:deprecated="true" hpcc:hidden="true" type="xs:boolean" use="optional"/>
-                            <xs:attribute name="state" hpcc:deprecated="true" hpcc:hidden="true" type="xs:string" use="optional"/>
-                            <xs:attribute name="trustedPeers" hpcc:deprecated="true" hpcc:hidden="true" type="xs:string" use="optional"/>
+                            <xs:attribute name="city" hpcc:deprecated="true" hpcc:hidden="true" type="xs:string"
```

Comment:
Why do you need  hpcc:deprecated? Why not remove the attribute?

Reply:
@wangkx Honestly, the concept of deprecated is new. No one has really figured out what it means, so for now, it is a way to mark a field as such. Removing the attribute would cause it to show up in the UI no matter what (the default behavior for attributes that are not defined in the schema per requirements). Leaving it would allow it to be marked hidden, so it has been left there for now.